python/cudaq/kernel/ast_bridge.py

```diff
@@ -461,15 +606,21 @@ def pushValue(self, value):
         visit method.
         """
         self.debug_msg(lambda: f'push {value}')
-        self.valueStack.append(value)
+        self.valueStack.pushValue(value)
 
     def popValue(self):
         """
         Pop an MLIR Value from the stack. 
         """
-        val = self.valueStack.pop()
+        val = self.valueStack.popValue()
         self.debug_msg(lambda: f'pop {val}')
         return val
+    
+    def popAllValues(self, expectedNumVals):
+        values = [self.popValue() for _ in range(self.valueStack.currentNumValues)]
+        if len(values) != expectedNumVals:
+            self.emitFatalError("processing error - no valid value was created", self.currentNode)
```

Comment:
nit: the error message here is not very clear. I think it should mention the mismatch between the number of values on the stack vs. the expected.